src/app/navbar/navbar.component.ts

```diff
@@ -37,8 +43,17 @@ export class NavComponent {
     }
 
     submitEmail() {
-        console.log(this.emailId);
-        this.toggleSubscribeBlock();
+        let headers = new Headers();
+        headers.append('Content-Type', 'application/json');
+        let options = new RequestOptions();
+        options.headers = headers;
+        this.http.post(this.globalConstants.getSubscribeAPIUrl(), `"${this.emailId}"`, options).toPromise().then(e => {
+            this.toggleSubscribeBlock();
```

Comment:
we can move this call outside the http call

Reply:
the correct thing to do is to show an error and not toggle if it fails. That's why I moved inside. Let's keep it like this and start showing error soon